### PR TITLE
testing/burp: new aport

### PR DESCRIPTION
**Pull request by BernhardGruen**
http://burp.grke.org
Burp is a network backup and restore program.

### CODE REVIEW

File: testing/burp/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	make DESTDIR="$pkgdir" install-all
+	rm -rf "$pkgdir"/var
+	rm -rf "$pkgdir"/etc/burp/autoupgrade
+	rm -rf "$pkgdir"/etc/burp/clientconfdir
+	chmod -R go-rwx "$pkgdir"/etc/burp
+}
+
+server() {
+	pkgdesc="burp is a network backup and restore program - server configuration and helper scripts"
+	depends="bash"
```
Comment (clandmeter) [inline testing/burp/APKBUILD]: Looks like you added bash for burp_ca which is a rather simple shell script. Would be nice if you could check/remove bash-ism and upstream a patch.

File: testing/burp/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	mkdir -p "$subpkgdir"/var/spool/burp
+	chmod 0755 "$subpkgdir"/var/spool
+	chmod 0700 "$subpkgdir"/var/spool/burp
+	mkdir -p "$subpkgdir"/usr/share/burp/scripts
+	mv "$pkgdir"/usr/share/burp/scripts "$subpkgdir"/usr/share/burp
+	rm -rf "$pkgdir"/usr/share
+
+	mkdir -p "$subpkgdir"/etc/burp
+	mv "$pkgdir"/etc/burp/CA.cnf "$subpkgdir"/etc/burp/CA.cnf
+	mv "$pkgdir"/etc/burp/burp-server.conf "$subpkgdir"/etc/burp/burp-server.conf
```
Comment (clandmeter) [inline testing/burp/APKBUILD]: Please try to keep max 80 char per line.

**Comment by BernhardGruen** [comment]
@clandmeter 
I will contact the author to ask him if he needs bash for other things. If he accepts to switch to /bin/sh I can support him in doing so.

File: testing/burp/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+arch="all"
+license="AGPL3"
+depends=""
+depends_dev=""
+makedepends="uthash-dev libressl-dev zlib-dev librsync-dev acl-dev"
+install=""
+subpackages="$pkgname-doc $pkgname-server"
+source="https://sourceforge.net/projects/$pkgname/files/$pkgname-$pkgver/$pkgname-$pkgver.tar.bz2 burp.init"
+builddir="$srcdir/$pkgname-$pkgver"
+options="!check"
```
[the remark this answers is not in the document]
Comment (BernhardGruen) [reply testing/burp/APKBUILD]: I will try to get that test suite running on alpine. It already runs on arch. Seems to be a problem with musl as it happens in the linker phase.

File: testing/burp/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+pkgdesc="Burp is a network backup and restore program"
+url="http://burp.grke.org"
+arch="all"
+license="AGPL3"
+depends=""
+depends_dev=""
+makedepends="uthash-dev libressl-dev zlib-dev librsync-dev acl-dev"
+install=""
+subpackages="$pkgname-doc $pkgname-server"
+source="https://sourceforge.net/projects/$pkgname/files/$pkgname-$pkgver/$pkgname-$pkgver.tar.bz2 burp.init"
```
[the remark this answers is not in the document]
Comment (BernhardGruen) [reply testing/burp/APKBUILD]: Done. also for other fields that have more than one entry (like makedepends).

File: testing/burp/APKBUILD (hunk 1 above)
Comment (BernhardGruen) [reply testing/burp/APKBUILD]: There are more scripts in /usr/share/burp/scripts that use bash. I will contact the author.

File: testing/burp/APKBUILD (hunk 2 above)
Comment (BernhardGruen) [reply testing/burp/APKBUILD]: Done.

File: testing/burp/APKBUILD [hunk 5]
```diff
@@ -14,6 +14,8 @@ makedepends="
 	zlib-dev
 	librsync-dev
 	acl-dev
+	libc6-compat
+	check-dev
```
Comment (clandmeter) [inline testing/burp/APKBUILD]: If these are purely for check phase you should add them to checkdepends. I do wonder why you need libc6-compat?

**Comment by BernhardGruen** [comment]
I can and will change the bash scripts to posix scripts. The original author most likely will accept my proposed changes.

The unit tests are something different. I don't find a way to get them pass. I don't know where I should search for a possible fix: check (check_pack.c line 346), musl (as it passes with glibc) or burp itself.

**Comment by BernhardGruen** [comment]
What is needed to get burp into testing?

**Comment by BernhardGruen** [comment]
I just squashed all commits into one and also updated to version 2.1.30 of burp.

**Pull request closed without merging**